code/projects/app/src/app/shared/image-display/image-display.component.ts

```diff
@@ -1,20 +1,69 @@
-import { Component, effect, ElementRef, input, viewChild } from "@angular/core";
+import {
+  animate,
+  state,
+  style,
+  transition,
+  trigger,
+} from "@angular/animations";
+import {
+  Component,
+  effect,
+  ElementRef,
+  input,
+  signal,
+  viewChild,
+} from "@angular/core";
 import { drawOn } from "../../util/image-data";
+import { ImageComparisonService } from "../image-comparison/image-comparison.service";
+
+type Flash = "red" | "green" | "default";
 
 @Component({
   selector: "app-image-display",
   templateUrl: "./image-display.component.html",
   styleUrl: "./image-display.component.css",
+  animations: [
+    trigger("flash", [
+      state("red", style({ backgroundColor: "var(--app-red)" })),
+      state("green", style({ backgroundColor: "var(--app-green)" })),
+      state("*", style({ backgroundColor: "var(--app-navy)" })),
+      transition("red => *", animate("300ms")),
+      transition("green => *", animate("300ms")),
+      transition("* => red", animate("0ms")),
+      transition("* => green", animate("0ms")),
+    ]),
+  ],
 })
 export class ImageDisplayComponent {
   public readonly image = input<ImageData>();
+  public showButtons = input<boolean>(true);
+  protected showControls = signal(false);
 
   private readonly canvas =
     viewChild.required<ElementRef<HTMLCanvasElement>>("canvas");
 
-  public constructor() {
+  protected state = { left: "default" as Flash, right: "default" as Flash };
+
+  constructor(private readonly imageLocalStorage: ImageComparisonService) {
     effect(() => {
       drawOn(this.image(), this.canvas().nativeElement);
+      this.showControls.set(this.image() !== undefined && this.showButtons());
```

Comment:
Declaring dependent state using `effect` is a huge anti-pattern, I'll leave an excerpt from the documentation at the bottom of this comment.

Instead, `showControls` could be declared as such:

```TS
protected readonly showControls = computed(() => this.image() !== undefined && this.showButtons());
```

`effect` is generally only used for side effects, so it is needed here to access the native DOM APIs, on the line above.

![obraz](https://github.com/user-attachments/assets/60493ce9-adc5-404b-8440-c995ae845dcf)